app/models/user.rb

```diff
@@ -1433,12 +1413,33 @@ def roles
     # array, so that we can append roles which are not yet migrated to the new system. This can be
     # removed once all roles are migrated to the new system.
 
-    if delegate_status.present?
-      roles << delegate_role
-    end
-
     roles.concat(team_roles)
 
     roles
   end
+
+  private def highest_delegate_role
+    highest_status_rank = nil
+    highest_role = nil
+    delegate_roles.each do |role|
+      status_rank = Api::V0::UserRolesController.status_sort_rank(role)
```

Comment:
Veto about accessing a controller from a model.
If really necessary, then the controller should access the model.

Reply:
Okay moved to UserRole model.